### PR TITLE
feat: typegen support for passing `Uint8Array` as `Vec<u8>`

### LINKED ISSUE #1553: Should accept Uint8Array as Array / data or bytes type
It's a pain having to convert between all kinds of array / bytes types. We should consider a connonical way to encode/decode between data and accept common forms of data like Uint8Arrays.

> [!NOTE]
> - Incoming from https://github.com/FuelLabs/fuels-ts/discussions/1545

## Steps

- [x] Add runtime support for passing `Uint8Array` as `Vec<8>` (#2018)
- [ ] Add typegen support for passing `Uint8Array` as `Vec<u8>`
- [ ] ...more?

### PR DESCRIPTION
**Pull request by Dhaiwat10**
- Closes #1553 

# Release notes

In this release, we:

- Added typegen support for passing `Uint8Array` as `Vec<u8`

# Summary

This PR improves typegen support for the `u8` type by adding `Uint8Array` as a valid input type on the function signature alongside `Vec<BigNumberish>`.

# Checklist

- [x] I **_added_** — `tests` to prove my changes
- [x] I **_updated_** — all the necessary `docs`
- [x] I **_reviewed_** — the entire PR myself, using the GitHub UI
- [ ] I **_described_** — all breaking changes and the Migration Guide

### CODE REVIEW

File: packages/fuel-gauge/src/vectors.test.ts [hunk 1]
```diff
@@ -34,6 +34,19 @@ describe('Vector Tests', () => {
     expect(value).toStrictEqual(INPUT);
   });
 
+  it('should test u8 vector input/output as Uint8Array', async () => {
+    using contractInstance = await setupContract();
+    const INPUT = [8, 6, 7, 5, 3, 0, 9];
+    const INPUT_AS_UINT8_ARRAY = Uint8Array.from(INPUT);
```
Comment (petertonysmith94) [inline packages/fuel-gauge/src/vectors.test.ts:40]:
```suggestion
    const INPUT: Vec<BigNumberish> = [8, 6, 7, 5, 3, 0, 9];
    const INPUT_AS_UINT8_ARRAY: Vec<BigNumberish> = Uint8Array.from(INPUT);
```

Is it worth adding the types here, to ensure they work as intended?

**Review by petertonysmith94 — COMMENTED** [review]
I'm experiencing some type issues with the boolean inputs?

image

---

Would it be worthwhile to add similar tests for `u16`, `u32`, ...?

File: packages/fuel-gauge/src/vectors.test.ts (hunk 1 above)
Comment (Dhaiwat10) [reply packages/fuel-gauge/src/vectors.test.ts:40]: @petertonysmith94 there's a separate issue for those types #2785, but I'm gonna be making changes to this PR regardless: https://github.com/FuelLabs/fuels-ts/pull/2884#discussion_r1704028350

**Comment by Dhaiwat10** [comment]
Closing since #1553 is not planned for now

**Pull request closed without merging**